metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/configuration/MetadataImporterMainApplicationConfiguration.java

```diff
@@ -18,47 +18,84 @@
 import org.springframework.context.annotation.Configuration;
-import org.springframework.core.convert.converter.Converter;
 import uk.ac.ebi.ampt2d.metadata.importer.ObjectsImporter;
-import uk.ac.ebi.ampt2d.metadata.importer.SraObjectsImporter;
-import uk.ac.ebi.ampt2d.metadata.importer.SraRetrieverByAccession;
+import uk.ac.ebi.ampt2d.metadata.importer.SraXmlRetrieverByAccession;
+import uk.ac.ebi.ampt2d.metadata.importer.api.SraObjectsImporterThroughAPI;
+import uk.ac.ebi.ampt2d.metadata.importer.converter.AnalysisConverter;
 import uk.ac.ebi.ampt2d.metadata.importer.converter.StudyConverter;
+import uk.ac.ebi.ampt2d.metadata.importer.database.SraObjectsImporterThroughDatabase;
+import uk.ac.ebi.ampt2d.metadata.importer.extractor.FileExtractorFromAnalysis;
 import uk.ac.ebi.ampt2d.metadata.importer.extractor.PublicationExtractorFromStudy;
 import uk.ac.ebi.ampt2d.metadata.importer.extractor.TaxonomyExtractor;
 import uk.ac.ebi.ampt2d.metadata.importer.extractor.WebResourceExtractorFromStudy;
+import uk.ac.ebi.ampt2d.metadata.importer.xml.SraAnalysisXmlParser;
 import uk.ac.ebi.ampt2d.metadata.importer.xml.SraStudyXmlParser;
-import uk.ac.ebi.ampt2d.metadata.importer.xml.SraXmlParser;
-import uk.ac.ebi.ampt2d.metadata.persistence.entities.Study;
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.FileRepository;
 import uk.ac.ebi.ampt2d.metadata.persistence.repositories.PublicationRepository;
 import uk.ac.ebi.ampt2d.metadata.persistence.repositories.TaxonomyRepository;
 import uk.ac.ebi.ampt2d.metadata.persistence.repositories.WebResourceRepository;
-import uk.ac.ebi.ena.sra.xml.StudyType;
 
 @Configuration
 public class MetadataImporterMainApplicationConfiguration {
 
     @Bean
-    public ObjectsImporter objectImporter(SraRetrieverByAccession sraRetrieverByAccession,
-                                          Converter<StudyType, Study> studyConverter,
-                                          PublicationRepository publicationRepository,
-                                          WebResourceRepository webResourceRepository) {
-        return new SraObjectsImporter(sraRetrieverByAccession, sraStudyXmlParser(), studyConverter,
-                new PublicationExtractorFromStudy(publicationRepository),
-                new WebResourceExtractorFromStudy(webResourceRepository));
+    @ConditionalOnProperty(name = "import.source", havingValue = "API")
+    public ObjectsImporter objectImporterForNonEga(SraXmlRetrieverByAccession sraXmlRetrieverByAccession,
+                                                   PublicationRepository publicationRepository,
+                                                   WebResourceRepository webResourceRepository,
+                                                   FileRepository fileRepository,
+                                                   TaxonomyRepository taxonomyRepository) {
+        return new SraObjectsImporterThroughAPI(sraXmlRetrieverByAccession, sraStudyXmlParser(),
```

Comment:
Sorry, I kinda lost here, I have implemented just passing the service layers to the Importer rather than the repository. the repositories are just for the configuration of service classes.